src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -169,6 +170,41 @@ public function get(): ?array
             }
         }
 
+        try {
+            // This could soon be optimized with https://github.com/antirez/redis/issues/5212 or
+            // https://github.com/antirez/redis/issues/6256
+            $pendingMessages = $this->connection->xpending($this->stream, $this->group, '-', '+', 3);
+        } catch (\RedisException $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+
+        $claimableIds = [];
+        foreach ($pendingMessages as $pendingMessage) {
+            if ($pendingMessage[1] === $this->consumer) {
+                $this->couldHavePendingMessages = true;
+                break;
+            }
+
+            if ($pendingMessage[2] > $this->claimTimeout) {
+                $claimableIds[] = $pendingMessage[0];
+            }
+        }
+
+        if (\count($claimableIds) > 0) {
+            try {
+                $this->connection->xclaim(
+                    $this->stream,
+                    $this->group,
+                    $this->consumer,
+                    $this->claimTimeout,
+                    $claimableIds,
+                    ['JUSTID']
+                );
```

Comment:
Is here not missing that we read the new claimed messages by setting `couldHavePendingMessages = true;` else it will not read the new claimed message in the next run.

Reply:
You're right. I thought this would lead to a race condition but I think the worst that can happen is that another consumer claimed the message in the meantime and `XREADGROUP` will return no message.